modules/flowable-inmemory-data/src/main/java/org/flowable/engine/data/inmemory/util/QueryUtil.java

```diff
@@ -0,0 +1,389 @@
+        }
+
+        switch (op) {
+        case EQUALS:
+        case EQUALS_IGNORE_CASE:
+            return nullSafeEquals(requested, actual);
+        case EXISTS:
+            return actual != null;
+        case LIKE:
+        case LIKE_IGNORE_CASE:
+            return requested.contains(stripLike(actual));
+        case NOT_EQUALS:
+        case NOT_EQUALS_IGNORE_CASE:
+            return !nullSafeEquals(requested, actual);
+        case NOT_EXISTS:
+            return actual == null;
+        case GREATER_THAN:
+        case GREATER_THAN_OR_EQUAL:
+        case LESS_THAN:
+        case LESS_THAN_OR_EQUAL:
+            return false;
+        default:
+            throw new IllegalStateException("Attempt to filter Text variable with unknown operator " + op);
+        }
+    }
+
+    private static boolean matchLongVariable(QueryOperator op, Long requested, Long actual) {
+        switch (op) {
+        case EQUALS:
+        case EQUALS_IGNORE_CASE:
+            if (actual == null) {
+                return false;
+            }
+            return requested.longValue() == actual.longValue();
+        case EXISTS:
+            return actual != null;
+        case GREATER_THAN:
+            if (actual == null) {
+                return false;
+            }
+            return actual.longValue() > requested.longValue();
+        case GREATER_THAN_OR_EQUAL:
+            if (actual == null) {
+                return false;
+            }
+            return actual.longValue() >= requested.longValue();
+        case LESS_THAN:
+            if (actual == null) {
+                return false;
+            }
+            return actual.longValue() < requested.longValue();
+        case LESS_THAN_OR_EQUAL:
+            if (actual == null) {
+                return false;
+            }
+            return actual.longValue() <= requested.longValue();
+        case LIKE:
+        case LIKE_IGNORE_CASE:
+            return false;
+        case NOT_EQUALS:
+        case NOT_EQUALS_IGNORE_CASE:
+            if (actual == null) {
+                return true;
+            }
+            return requested.longValue() != actual.longValue();
+        case NOT_EXISTS:
+            return actual == null;
+        default:
+            throw new IllegalStateException("Attempt to filter Long variable with unknown operator " + op);
+        }
+    }
+
+    private static boolean matchDoubleVariable(QueryOperator op, Double requested, Double actual) {
+        switch (op) {
+        case EQUALS:
+        case EQUALS_IGNORE_CASE:
+            if (actual == null) {
+                return false;
+            }
+            return requested.doubleValue() == actual.doubleValue();
+        case EXISTS:
+            return actual != null;
+        case GREATER_THAN:
+            if (actual == null) {
+                return false;
+            }
+            return actual.doubleValue() > requested.doubleValue();
```

Comment:
why is boxing of the value being done?  the values can be compared directly especially starting with JDK 5.

Reply:
removed unnecessary unboxing